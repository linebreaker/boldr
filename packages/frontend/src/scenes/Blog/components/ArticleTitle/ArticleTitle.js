/* @flow */
import React from 'react';
import styled from 'styled-components';
import cx from 'classnames';
import Headline from '@boldr/ui/Headline';
import { StyleClasses } from '../../../../theme/styleClasses';

const TitleWrapper = styled.div`
  text-transform: uppercase;
  padding-top: 150px;
  text-align: center;
  margin: 0 auto;
`;

const BASE_ELEMENT = StyleClasses.ARTICLE_TITLE;

const ArticleTitle = (props: { title: string, className?: string }) => {
  const classes = cx(BASE_ELEMENT, props.className);
  return (
    <TitleWrapper>
      <Headline type="h1" className={classes} lightText>
        {props.title}
      </Headline>
    </TitleWrapper>
  );
};

export default ArticleTitle;
