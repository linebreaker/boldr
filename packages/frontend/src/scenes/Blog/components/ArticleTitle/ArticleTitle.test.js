import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ArticleTitle from './ArticleTitle';

describe('<ArticleTitle />', () => {
  it('accepts props and renders them.', () => {
    const wrapper = shallow(<ArticleTitle title="Random Post Title" />);
    expect(wrapper.instance().props.title).toBe('Random Post Title');
  });
  it('renders snapshot', () => {
    const wrapper = shallow(<ArticleTitle created="Random Post Title" />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
