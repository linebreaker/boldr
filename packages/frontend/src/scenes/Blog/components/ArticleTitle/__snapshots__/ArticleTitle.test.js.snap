// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleTitle /> renders snapshot 1`] = `
<ArticleTitle__TitleWrapper>
  <Headline
    className="boldrui-article__title"
    lightText={true}
    type="h1"
  />
</ArticleTitle__TitleWrapper>
`;
